From 8252dd6fdbaf4572097991abb8e2b93e83e538bf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Kochma=C5=84ski?= <daniel@turtleware.eu>
Date: Thu, 9 Feb 2017 22:07:05 +0000
Subject: [PATCH] Merge branch 'patch-1' into 'develop'

In `flisten` check that the stream is not in an error state, as well as not eof

See merge request !51
---
 src/c/file.d | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/c/file.d b/src/c/file.d
index 0f739b0..2122485 100755
--- a/src/c/file.d
+++ b/src/c/file.d
@@ -5314,7 +5314,7 @@ static int
 flisten(cl_object stream, FILE *fp)
 {
   int aux;
-  if (feof(fp))
+  if (feof(fp) || ferror(fp))
     return ECL_LISTEN_EOF;
 #ifdef FILE_CNT
   if (FILE_CNT(fp) > 0)
-- 
1.9.1

